code/game/objects/structures/guillotine.dm

```diff
@@ -0,0 +1,229 @@
+
+	if(has_buckled_mobs())
+		msg += "<br/>"
+		msg += "Someone appears to be strapped in. You can help them out, or you can harm them by activating the guillotine."
+
+	to_chat(user, msg)
+
+	return msg
+
+/obj/structure/guillotine/attack_hand(mob/user)
+	add_fingerprint(user)
+
+	// Currently being used by something
+	if(current_action)
+		return
+
+	switch(blade_status)
+		if(GUILLOTINE_BLADE_MOVING)
+			return
+		if(GUILLOTINE_BLADE_DROPPED)
+			blade_status = GUILLOTINE_BLADE_MOVING
+			icon_state = "guillotine_raise"
+			addtimer(src, "raise_blade", GUILLOTINE_ANIMATION_LENGTH)
+			return
+		if(GUILLOTINE_BLADE_RAISED)
+			if(has_buckled_mobs())
+				if(user.a_intent == INTENT_HARM)
+					user.visible_message("<span class='warning'>[user] begins to pull the lever!</span>",
+						                 "<span class='warning'>You begin to the pull the lever.</span>")
+					current_action = GUILLOTINE_ACTION_INUSE
+
+					if(do_after(user, GUILLOTINE_ACTIVATE_DELAY, target = src) && blade_status == GUILLOTINE_BLADE_RAISED)
+						current_action = 0
+						blade_status = GUILLOTINE_BLADE_MOVING
+						icon_state = "guillotine_drop"
+						playsound(src, 'sound/items/unsheath.ogg', 100, 1)
+						addtimer(src, "drop_blade", GUILLOTINE_ANIMATION_LENGTH - 2, FALSE, user) // Minus two so we play the sound and decap faster
+					else
+						current_action = 0
+				else
+					unbuckle_mob()
+			else
+				blade_status = GUILLOTINE_BLADE_MOVING
+				icon_state = "guillotine_drop"
+				playsound(src, 'sound/items/unsheath.ogg', 100, 1)
+				addtimer(src, "drop_blade", GUILLOTINE_ANIMATION_LENGTH)
+
+/obj/structure/guillotine/proc/raise_blade()
+	blade_status = GUILLOTINE_BLADE_RAISED
+	icon_state = "guillotine_raised"
+
+/obj/structure/guillotine/proc/drop_blade(mob/user)
+	if(has_buckled_mobs() && blade_sharpness)
+		var/mob/living/carbon/human/H = buckled_mob
+
+		if(!H)
+			return
+
+		var/obj/item/organ/external/head/head = H.get_organ("head")
+
+		if(QDELETED(head))
+			return
+
+		playsound(src, 'sound/weapons/bladeslice.ogg', 100, 1)
+		if(blade_sharpness >= GUILLOTINE_DECAP_MIN_SHARP || head.brute_dam >= 100)
+			head.droplimb()
+			add_attack_logs(user, H, "beheaded with [src]")
+			H.regenerate_icons()
+			unbuckle_mob()
+			kill_count += 1
+
+			var/blood_overlay = "bloody"
+
+			if(kill_count == 2)
+				blood_overlay = "bloodier"
+			else if(kill_count > 2)
+				blood_overlay = "bloodiest"
+
+			blood_overlay = "guillotine_" + blood_overlay + "_overlay"
+			overlays.Cut()
+			overlays += mutable_appearance(icon, blood_overlay)
+
+			// The crowd is pleased
+			// The delay is to make large crowds have a longer lasting applause
+			var/delay_offset = 0
+			for(var/mob/living/carbon/human/HM in viewers(src, 7))
+				addtimer(HM, "emote", delay_offset * 0.3, FALSE, "clap")
+				delay_offset++
+		else
+			H.apply_damage(15 * blade_sharpness, BRUTE, head)
+			add_attack_logs(user, H, "dropped the blade on non-fatally with [src]")
```

Comment:
Probably should re-order these words to "non-fatally dropped the blade on" or something to that effect. Current wording is clunky as hell